--- a/chrome/browser/ui/views/bookmarks/bookmark_menu_delegate.cc
+++ b/chrome/browser/ui/views/bookmarks/bookmark_menu_delegate.cc
@@ -186,7 +186,7 @@ void BookmarkMenuDelegate::ExecuteCommand(int id, int mouse_event_flags) {
 bool BookmarkMenuDelegate::ShouldExecuteCommandWithoutClosingMenu(
     int id,
     const ui::Event& event) {
-  if ((event.flags() & ui::EF_LEFT_MOUSE_BUTTON) &&
+  if ((event.flags() & ui::EF_LEFT_MOUSE_BUTTON || event.flags() & ui::EF_MIDDLE_MOUSE_BUTTON) &&
       ui::DispositionFromEventFlags(event.flags()) ==
           WindowOpenDisposition::NEW_BACKGROUND_TAB) {
     DCHECK(menu_id_to_node_map_.find(id) != menu_id_to_node_map_.end());
@@ -200,7 +200,16 @@
     const BookmarkNode* node = menu_id_to_node_map_[id];
     // Close the menu before opening a folder since this may pop up a dialog
     // over the menu. See https://crbug.com/1105587 for details.
-    return node->type() != BookmarkNode::FOLDER;
+    if (node->type() != BookmarkNode::FOLDER)
+      return true;
+    size_t num_of_urls_in_folder = 0;
+    for (const auto& child : node->children()) {
+      if (child->is_url())
+        num_of_urls_in_folder++;
+      if (num_of_urls_in_folder == chrome::kNumBookmarkUrlsBeforePrompting)
+        return false;
+    }
+    return true;
   }
   return false;
 }
