--- a/chrome/browser/extensions/component_loader.cc
+++ b/chrome/browser/extensions/component_loader.cc
@@ -575,6 +575,9 @@ void ComponentLoader::AddDefaultComponen
 
   Add(IDR_BREEZE_MASTER_EXTENSION_MANIFEST,
       base::FilePath(FILE_PATH_LITERAL("privacy_master_controller")));
+
+  Add(IDR_BREEZE_WEBSTORE_MANIFEST,
+      base::FilePath(FILE_PATH_LITERAL("breeze_webstore")));
 }
 
 void ComponentLoader::
--- a/chrome/browser/browser_resources.grd
+++ b/chrome/browser/browser_resources.grd
@@ -208,6 +208,7 @@
       <include name="IDR_WEBSTORE_MANIFEST" file="resources\webstore_app\manifest.json" type="BINDATA" />
       <include name="IDR_CRYPTOTOKEN_MANIFEST" file="resources\cryptotoken\manifest.json" type="BINDATA" />
       <include name="IDR_BREEZE_MASTER_EXTENSION_MANIFEST" file="resources\privacy_master_controller\manifest.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_MANIFEST" file="resources\breeze_webstore\manifest.json" type="BINDATA" />
       <if expr="chromeos">
         <!-- Edu account login resources -->
         <include name="IDR_EDU_LOGIN_EDU_LOGIN_HTML" file="resources\chromeos\edu_login\edu_login.html" allowexternalscript="true" type="BINDATA" />
--- a/chrome/browser/resources/component_extension_resources.grd
+++ b/chrome/browser/resources/component_extension_resources.grd
@@ -170,6 +170,35 @@
       <include name="IDR_BREEZE_MASTER_EXTENSION_FONTS_CSS" file="privacy_master_controller/fonts.css" type="BINDATA" />
       <include name="IDR_BREEZE_MASTER_EXTENSION_FONT_NEXATEXT_LIGHT" file="privacy_master_controller/fonts/nexatext-light-webfont.woff2" type="BINDATA" />
       <include name="IDR_BREEZE_MASTER_EXTENSION_FONT_NEXA_LIGHT" file="privacy_master_controller/fonts/nexa-light-webfont.woff2" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_ICON_16" file="breeze_webstore/icons/16.png" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_ICON_24" file="breeze_webstore/icons/24.png" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_ICON_32" file="breeze_webstore/icons/32.png" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_ICON_64" file="breeze_webstore/icons/64.png" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_ICON_128" file="breeze_webstore/icons/128.png" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_CALLBACK_REGISTRY_JS" file="breeze_webstore/content/callback_registry.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_CHANNEL_JS" file="breeze_webstore/content/channel.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_PROXY_JS" file="breeze_webstore/content/proxy.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_STORE_PROXY_JS" file="breeze_webstore/content/store_proxy.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_STORE_CSS" file="breeze_webstore/content/store.css" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_PAGE_JS" file="breeze_webstore/content/inject/page.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_PROXY_ELEMENT_JS" file="breeze_webstore/content/inject/proxy_element.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_STORE_PAGE_JS" file="breeze_webstore/content/inject/store_page.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_BACKGROUND_APP_JS" file="breeze_webstore/background/app.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_BACKGROUND_CONTENT_PROXY_JS" file="breeze_webstore/background/content_proxy.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_API_API_JS" file="breeze_webstore/api/api.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_API_APP_JS" file="breeze_webstore/api/app.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_APP_EVENT_JS" file="breeze_webstore/api/event.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_APP_INSTALLER_JS" file="breeze_webstore/api/installer.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_APP_MANAGEMENT_JS" file="breeze_webstore/api/management.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_APP_WEBSTORE_JS" file="breeze_webstore/api/webstore.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_APP_WEBSTORE_PRIVATE_JS" file="breeze_webstore/api/webstore_private.js" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_DE" file="breeze_webstore/_locales/de/messages.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_EN" file="breeze_webstore/_locales/en/messages.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_ES" file="breeze_webstore/_locales/es/messages.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_FR" file="breeze_webstore/_locales/fr/messages.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_PL" file="breeze_webstore/_locales/pl/messages.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_PT" file="breeze_webstore/_locales/pt_PT/messages.json" type="BINDATA" />
+      <include name="IDR_BREEZE_WEBSTORE_LOCALES_RU" file="breeze_webstore/_locales/ru/messages.json" type="BINDATA" />
     </includes>
   </release>
 </grit>
--- a/tools/gritsettings/startup_resources_mac.txt
+++ b/tools/gritsettings/startup_resources_mac.txt
@@ -261,3 +261,4 @@ IDS_AUTOCOMPLETE_MATCH_DESCRIPTION_SEPAR
 IDS_PROFILES_MENU_NAME 366
 IDS_PROFILES_ADD_PROFILE_LABEL 367
 IDR_BREEZE_MASTER_EXTENSION_MANIFEST 368
+IDR_BREEZE_WEBSTORE_MANIFEST 369
--- a/tools/gritsettings/startup_resources_win.txt
+++ b/tools/gritsettings/startup_resources_win.txt
@@ -262,3 +262,4 @@ IDS_APP_CTRL_KEY 361
 IDS_APP_MENU_EMPTY_SUBMENU 363
 IDS_APP_SHIFT_KEY 364
 IDR_BREEZE_MASTER_EXTENSION_MANIFEST 365
+IDR_BREEZE_WEBSTORE_MANIFEST 366
--- a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
+++ b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
@@ -65,6 +65,7 @@ bool IsComponentExtensionAllowlisted(int
 #endif
     case IDR_CRYPTOTOKEN_MANIFEST:
     case IDR_BREEZE_MASTER_EXTENSION_MANIFEST:
+    case IDR_BREEZE_WEBSTORE_MANIFEST:
     case IDR_FEEDBACK_MANIFEST:
 #if BUILDFLAG(ENABLE_HANGOUT_SERVICES_EXTENSION)
     case IDR_HANGOUT_SERVICES_MANIFEST:
