--- a/chrome/app/profiles_strings.grdp
+++ b/chrome/app/profiles_strings.grdp
@@ -76,3 +76,3 @@
   <message name="IDS_PROFILES_OPTIONS_GROUP_NAME" desc="The name of the sync group in the options dialog when using the new Profiles UI.">
-    People
+    Profiles
   </message>
@@ -80,3 +80,3 @@
     <message name="IDS_PROFILES_CREATE_NEW_PROFILE_OPTION" desc="In Title Case. Create new profile menu item and button title.">
-      Add Person...
+      Add Profile
     </message>
@@ -142,3 +142,3 @@
     <message name="IDS_PROFILES_OTHER_PROFILES_TITLE" desc="Title in the profile menu of the 'other profiles' section.">
-      Other people
+      Other profiles
     </message>
@@ -151,3 +151,3 @@
     <message name="IDS_PROFILES_MANAGE_USERS_BUTTON" desc="Button in the avatar menu bubble view for launching the user manager.">
-      Manage people
+      Manage profiles
     </message>
@@ -156,3 +156,3 @@
     <message name="IDS_PROFILES_OTHER_PROFILES_TITLE" desc="Title in the profile menu of the 'other profiles' section.">
-      Other People
+      Other Profiles
     </message>
@@ -165,3 +165,3 @@
     <message name="IDS_PROFILES_MANAGE_USERS_BUTTON" desc="Button in the avatar menu bubble view for launching the user manager.">
-      Manage People
+      Manage Profiles
     </message>
@@ -187,3 +187,3 @@
   <message name="IDS_NEW_NUMBERED_PROFILE_NAME" desc="The name given to a newly created profile. Displayed in wrench menu and settings UI using the --new-profile-management flag.">
-    Person <ph name="VALUE">$1<ex>1</ex></ph>
+    Profile <ph name="VALUE">$1<ex>1</ex></ph>
   </message>
@@ -425,6 +425,6 @@
   <message name="IDS_PROFILES_CREATE_BUTTON_LABEL" desc="The label that appears on the profile create button in the options dialog to allow a user to create a new profile.">
-    Add person...
+    Add profile
   </message>
   <message name="IDS_PROFILES_MANAGE_BUTTON_LABEL" desc="The label that appears on the profile manage button in the options dialog to allow a user to manage their profile.">
-    Edit...
+    Edit profile
   </message>
@@ -446,6 +446,6 @@
   <message name="IDS_OLD_PROFILES_DISABLED_ADD_PERSON_SUGGESTION" desc="The suggestion that tells the user to create a new profile when the user tries to use an old profile that is disabled by policy">
-    To continue, click OK, then click Add Person to create a new profile for your email address.
+    To continue, click OK, then click Add Profile to create a new profile for your email address.
   </message>
   <message name="IDS_OLD_PROFILES_DISABLED_ADD_PERSON_SUGGESTION_WITH_DOMAIN" desc="The suggestion that tells the user to create a new profile with the domain when the user tries to use an old profile that is disabled by policy">
-    To continue, click OK, then click Add Person to create a new profile for your <ph name="DOMAIN">$1<ex>example.com</ex></ph> email address.
+    To continue, click OK, then click Add Profile to create a new profile for your <ph name="DOMAIN">$1<ex>example.com</ex></ph> email address.
   </message>
@@ -472,3 +472,3 @@
     <message name="IDS_PROFILES_CREATE_TITLE" desc="Title of the create profile dialog">
-      Add person
+      Add profile
     </message>
@@ -506,3 +506,3 @@
       <message name="IDS_ADD_USER_BUTTON" desc="Text shown on an add user button on login/locker screen">
-        Add Person
+        Add Profile
       </message>
@@ -510,6 +510,6 @@
         <message name="IDS_LOGIN_POD_USER_REMOVE_WARNING_BUTTON" desc="Text shown on a button that confirms removal of a user.">
-          Remove This Person
+          Remove This Profile
         </message>
         <message name="IDS_CREATE_LEGACY_SUPERVISED_USER_MENU_LABEL" desc="Text shown on a menu option that creates a legacy supervised user.">
-          Add Supervised Person
+          Add Supervised Profile
         </message>
@@ -530,3 +530,3 @@
       <message name="IDS_ADD_USER_BUTTON" desc="Text shown on an add user button on login/locker screen">
-        Add person
+        Add profile
       </message>
@@ -534,6 +534,6 @@
         <message name="IDS_LOGIN_POD_USER_REMOVE_WARNING_BUTTON" desc="Text shown on a button that confirms removal of a user.">
-          Remove this person
+          Remove this profile
         </message>
         <message name="IDS_CREATE_LEGACY_SUPERVISED_USER_MENU_LABEL" desc="Text shown on a menu option that creates a legacy supervised user.">
-          Add supervised person
+          Add supervised profile
         </message>
@@ -627,6 +627,6 @@
       <message name="IDS_USER_MANAGER_REMOVE_PROFILE_PROFILES_LOCKED_ERROR" desc="Error message displayed when trying to remove a profile while all profiles are locked.">
-        Please unlock your profile before removing a person.
+        Please unlock your profile before removing a profile.
       </message>
       <message name="IDS_USER_MANAGER_ADD_PROFILE_PROFILES_LOCKED_ERROR" desc="Error message displayed when trying to add a profile while all profiles are locked.">
-        Please unlock your profile before adding a person.
+        Please unlock your profile before adding a new profile.
       </message>
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -2648,3 +2648,3 @@
   <message name="IDS_SETTINGS_EDIT_PERSON" desc="Title of the edit person subpage. This is a subpage to edit the name and icon of a Chrome Profile.">
-    Edit person
+    Edit profile
   </message>
